docs/sources/flow/reference/components/otelcol.processor.attributes.md

```diff
@@ -0,0 +1,704 @@
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`key` | `string` | The attribute that the action relates to. |  | yes
+`action` | `string` | The type of action to be performed. |  | yes
+`value` | `any` | The value to populate for the key. |  | no
+`pattern` | `string` | A regex pattern. | `""` | no
+`from_attribute` | `string` | The attribute from the input data to use to populate the attribute value. | `""` | no
+`from_context` | `string` | The context value to use to populate the attribute value.  | `""` | no
+`converted_type` | `string` | The type to convert the attribute value to. | `""` | no
+
+The type of `value` must be either a number, string, or boolean.
+
+The supported values for `action` are:
+
+* `insert`: Inserts a new attribute in input data where the key does not already exist.
+
+    * The `key` attribute is required. It specifies the attribute to act upon.
+    * One of the `value`, `from_attribute` or `from_context` attributes is required.
+
+* `update`: Updates an attribute in input data where the key does exist.
+
+    * The `key`attribute is required. It specifies the attribute to act upon.
+    * One of the `value`, `from_attribute` or `from_context` attributes is required.
+
+* `upsert`: Either inserts a new attribute in input data where the key does not already exist, or updates an attribute in input data where the key does exist.
+
+    * The `key`attribute is required. It specifies the attribute to act upon.
+    * One of the `value`, `from_attribute` or `from_context`attributes is required:
+        * `value` specifies the value to populate for the key.
+        * `from_attribute` specifies the attribute from the input data to use to populate
+        the value. If the attribute doesn't exist, no action is performed.
+        * `from_context` specifies the context value to use to populate the attribute value.
+        If the key is prefixed with `metadata.`, the values are searched
+        in the receiver's transport protocol additional information like gRPC Metadata or HTTP Headers. 
+        If the key is prefixed with `auth.`, the values are searched
+        in the authentication information set by the server authenticator. 
+        Refer to the server authenticator's documentation part of your pipeline 
+        for more information about which attributes are available.
+        If the key doesn't exist, no action is performed.
+        If the key has multiple values the values will be joined with `;` separator.
+
+* `hash`: Hashes (SHA1) an existing attribute value.
+
+    * The `key` attribute and/or the `pattern` attributes is required.
+
+* `extract`: Extracts values using a regular expression rule from the input key to target keys specified in the rule. If a target key already exists, it will be overridden. Note: It behaves similar to the Span Processor `to_attributes` setting with the existing attribute as the source.
```

Comment:
```suggestion
* `extract`: Extracts values using a regular expression rule from the input key to target keys specified in the rule. If a target key already exists, it will be overridden. Note: It behaves similarly to the Span Processor `to_attributes` setting with the existing attribute as the source.
```